### PR TITLE
feat(form/linting): account for full semver specifiying version

### PR DESCRIPTION
**Pull request by nikku**
* Account for full `major.minor.patch` being serialized in diagrams.
* Show `major.minor` in error messages only

---

#### Built Artifacts

* https://camunda-modeler-on-demand.s3.eu-central-1.amazonaws.com/form-semver-maj-mi-pa/camunda-modeler-form-semver-maj-mi-pa-linux-x64.tar.gz
* https://camunda-modeler-on-demand.s3.eu-central-1.amazonaws.com/form-semver-maj-mi-pa/camunda-modeler-form-semver-maj-mi-pa-mac.dmg
* https://camunda-modeler-on-demand.s3.eu-central-1.amazonaws.com/form-semver-maj-mi-pa/camunda-modeler-form-semver-maj-mi-pa-mac.zip
* https://camunda-modeler-on-demand.s3.eu-central-1.amazonaws.com/form-semver-maj-mi-pa/camunda-modeler-form-semver-maj-mi-pa-win-ia32.zip
* https://camunda-modeler-on-demand.s3.eu-central-1.amazonaws.com/form-semver-maj-mi-pa/camunda-modeler-form-semver-maj-mi-pa-win-x64.zip

### CODE REVIEW

File: client/src/app/tabs/form/linting/__tests__/FormLinterSpec.js [hunk 1]
```diff
@@ -59,9 +61,11 @@ describe('FormLinter', function() {
   [
     [ 'Camunda Platform 7.15', camundaPlatform715, camundaPlatform715Errors ],
     [ 'Camunda Platform 7.16', camundaPlatform716 ],
+    [ 'Camunda Platform 7.16.0', camundaPlatform7160 ],
     [ 'Camunda Cloud 1.0', camundaCloud10, camundaCloud10Errors ],
     [ 'Camunda Cloud 1.1', camundaCloud11 ],
     [ 'Camunda Cloud 1.2', camundaCloud12 ],
+    [ 'Camunda Cloud 1.2', camundaCloud120 ],
```
Comment (philippfromme) [inline client/src/app/tabs/form/linting/__tests__/FormLinterSpec.js:68]:
```suggestion
    [ 'Camunda Cloud 1.2.0', camundaCloud120 ],
```

Comment (nikku) [reply client/src/app/tabs/form/linting/__tests__/FormLinterSpec.js:68]: Hence the test.

**Comment by nikku** [comment]
@andreasgeier I'll provide you with a distribution of this one + all linting related improvements. Stay tuned :clock1:

**Comment by nikku** [comment]
@philippfromme @andreasgeier Built artifacts available for testing the whole package :arrow_up:.

**Comment by andreasgeier** [comment]
If you realize unexpected behavior or make changes during implementation, please report it asap so that I can reconsider the proposal and how it affects UI/UX.

Change of order in status bar:
The linting status is placed between profile selection and deploy. I understand the logic behind this change since it reflects the usual workflow: select execution profile -> debug your diagram -> deploy it -> run it.
But I disagree with this change and request the order as it is defined in the prototype. The reason for this is that the text string of the linting output will constantly change (its length) which will cause a repositioning of the following icons for deploy + run every time. This might become an even bigger issue when we have warnings, too.
When we have warnings for the first time, I want to reconsider it and might find a solution with a static width which would allow us to put it in the logical order without flickering of the other symbols. For now, please stick to the design.

Unfriendly validation of tasks
I see that the creation of a task immediately results in a linting error. I will give this a thought because it is not a desired default behavior that users get permanent error messages while they model their process.
However, me might see the same behavior when we integrate implementation property linting. This is a result of not having a dedicated design and implementation phase as originally defined.
@nikku Did you give this a thought and have a proposal how to fix it?

**Comment by nikku** [comment]
> But I disagree with this change and request the order as it is defined in the prototype. 

Let's move it. 

----

> I see that the creation of a task immediately results in a linting error. I will give this a thought because it is not a desired default behavior that users get permanent error messages while they model their process.

As you mention, this is _by design_ at the moment:

* Users must model tasks first, change them to something executable later
* A task is shown as an error (which it is; it is not executable), rather than a warning

We could consider tuning in both directions. Looking into the future (i.e. connectors) it will be easier to place stuff that can just be executed right from the palette/context-pad.

For _modeling tasks first_ I'd argue that this is about a user understanding what is needed to make a process executable. The user learns what needs to be done, and that is a good thing. 

Mid-term _auto-fix_ capabilities could allow us to move through each of these errors, prompting the user what to do (i.e. how shall that task be implemented?). :arrow_right: This will be a massive improvement to the modeling UX, as it supports the "make an existing diagram executable" story. 

Short term, we could move to a execution focused palette (only showing actually executable elements in Camunda Cloud editor). But this is where we came from (and decided to ditch it, in favor of generic first). :arrow_right: I'd not go down that route but wait for feedback first. It is fine to implement something that is incomplete first (cf. also your point on missing properties; it will be a common use-case!).

---

CC @currandwyer

**Comment by andreasgeier** [comment]
Regarding validation of tasks in a cloud diagram:

I agree that we should keep it for now as it is: a task without type causes a compatibility error. This is part of a bigger question and is also related to upcoming implementation props linting.

I see some options how we could fix this in the future to avoid unfriendly validation for modeling-only use cases. For instance, let users toggle the linting off/on.

**Comment by andreasgeier** [comment]
Design review summary:

- change order of linting output in status bar
- change linting output color to red (see [prototype](https://fluffy-fiesta-212e1c7c.pages.github.io/prototypes/visual-ide/v16/))
- let's keep the task linting behavior as it is implemented

**Comment by nikku** [comment]
> I see some options how we could fix this in the future to avoid unfriendly validation for modeling-only use cases. For instance, let users toggle the linting off/on.

Exactly. Users could, at any time, decide which categories of errors to account for. If I'm modeling only I can disable linting for execution this way.

**Comment by andreasgeier** [comment]
I really like that the click-to-select action is already part of the linting feature ❤️

**Comment by philippfromme** [comment]
@andreasgeier I've adjusted the linting status bar item and updated the artifacts so you can see it in action.

Camunda_Modeler_SHCJFVeO53

**Pull request merged**